WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ReaderViewPage.java

```diff
@@ -0,0 +1,107 @@
+package org.wordpress.android.e2e.pages;
+
+import android.view.KeyEvent;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiSelector;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.wordpress.android.support.WPSupportUtils.DEFAULT_TIMEOUT;
+import static org.wordpress.android.support.WPSupportUtils.isTextDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.swipeToLeft;
+import static org.wordpress.android.support.WPSupportUtils.swipeToRight;
+
+public class ReaderViewPage {
+    UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+    String mLikerContainerId = "org.wordpress.android.prealpha:id/liker_faces_container";
+    String mRelatedPostsId = "org.wordpress.android.prealpha:id/container_related_posts";
+
+    UiObject mLikerContainer = mDevice.findObject(new UiSelector().resourceId(mLikerContainerId));
+    UiObject mRelatedPostsContainer = mDevice.findObject(new UiSelector().resourceId(mRelatedPostsId));
+    UiObject mSwipeForMore = mDevice.findObject(new UiSelector().textContains("Swipe for more"));
+
+    public ReaderViewPage waitUntilLoaded() {
+        mRelatedPostsContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage like() {
+        tapLikeButton();
+        mLikerContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage unlike() {
+        tapLikeButton();
+        mLikerContainer.waitUntilGone(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    private void tapLikeButton() {
+        mSwipeForMore.waitUntilGone(DEFAULT_TIMEOUT);
+        // Even though it was working locally in simulator, tapping the footer buttons,
+        // like 'mLikeButton.click()', was not working in CI.
+        // The current workaround is to use arrows navigation.
+
+        // Bring focus to the footer. First button is selected.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_DOWN);
+        // Navigate to Like button.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        // Click the Like button.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_CENTER);
+        // Navigate back to the first footer button.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
```

Comment:
I will not even start suggesting anything about this deep magic. I believe you spent enough time with CI to be sure that this is the way.

Reply:
😄 Hey, 'this is the way' is too strong. That's the only **_temporary_** workaround I've found that worked in CI.